### PR TITLE
Feat/#341 Delete Confirmation Modal

### LINKED ISSUE #341: Confirmation Modal for Deleting Message
As an EmbeddedChat developer

**I need to**: Implement a confirmation modal for deleting the message, just like for 'Report Message'. 

**So That**: A confirmation is asked before deleting, and the Message is not deleted by mistake.

**Acceptance Criteria**

- [X] Confirmation modal for message-delete.
- [X] With Cancel, and Delete Message options.

### PR DESCRIPTION
**Pull request by umangutkarsh**
# Brief Title
Confirm Modal implementation to delete message, just like for 'report-message' is implemented, so that the message is not deleted by mistake.

## Acceptance Criteria fulfillment

- [X] Implement delete message modal.
- [X] Use modal in `MessageToolbox.js`  

Fixes #341 

## Video/Screenshots
[delete.webm](https://github.com/RocketChat/EmbeddedChat/assets/95426993/2017dd2c-082a-4199-b46a-5103b6750671)

### CODE REVIEW

File: packages/react/src/components/Message/MessageToolbox.js [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
 import { Box } from '../Box';
 import { appendClassNames } from '../../lib/appendClassNames';
 import { ActionButton } from '../ActionButton';
 import { EmojiPicker } from '../EmojiPicker';
+import { Modal } from '../Modal';
+import { Icon } from '../Icon';
+import { Button } from '../Button';
+import RCContext from '../../context/RCInstance';
+import { useToastBarDispatch } from '../../hooks/useToastBarDispatch';
+import { useToastStore } from '../../store';
```
Comment (abhinavkrin) [inline packages/react/src/components/Message/MessageToolbox.js:14]: This was removed recently and we are no longer using the store for toasts. I think this is the older code.

File: packages/react/src/components/Message/MessageToolbox.js [hunk 2]
```diff
@@ -54,88 +59,145 @@ export const MessageToolbox = ({
...
     style
   );
+
+  const { RCInstance } = useContext(RCContext);
+  const dispatchToastMessage = useToastBarDispatch();
+  const toastPosition = useToastStore((state) => state.position);
+
   const [isEmojiOpen, setEmojiOpen] = useState(false);
+
+  const [deleteId, setDeleteId] = useState('');
```
Comment (abhinavkrin) [inline packages/react/src/components/Message/MessageToolbox.js:69]: deleteId is state is never used.

**Review by abhinavkrin — CHANGES_REQUESTED** [review]

**Comment by umangutkarsh** [comment]
Hey. Thanks for reviewing. Will address the changes asap. 
Thanks.

File: packages/react/src/components/Message/MessageToolbox.js [hunk 3]
```diff
@@ -54,88 +59,145 @@ export const MessageToolbox = ({
...
+  const handleOnClose = () => {
+    setShowDeleteModal(false);
+  };
+
+  const handleClickDelete = (message) => {
+    setDeleteId(message._id);
+    setShowDeleteModal(true);
+  };
+
+  const handleDeleteMessage = async (message) => {
```
[the remark this answers is not in the document]
Comment (umangutkarsh) [reply packages/react/src/components/Message/MessageToolbox.js]: Moved the `handleDeleteMessage()` to it's old location and used it as a prop.

**Comment by umangutkarsh** [comment]
Hi @abhinavkrin . I've made the changes, there were some conflicts with the develop branch that's why the `toastStore`  was being used. Please review the PR and if anything further is required, please let me know.
Thanks.

**Review by abhinavkrin — APPROVED** [review]
Thanks @umangutkarsh. This will prevent users from accidentally deleting a message. I have added a minor change in which the message preview is shown in the modal.
delete_modal

**Pull request merged**